tempesta_fw/client.c

```diff
@@ -28,122 +28,222 @@
 #include "connection.h"
 #include "log.h"
 #include "procfs.h"
+#include "tdb.h"
+#include "lib/str.h"
+#include "lib/common.h"
 
-#define CLI_HASH_BITS	17
-#define CLI_HASH_SZ	(1 << CLI_HASH_BITS)
+/* Length of comparison of clients entry by User-Agent. */
+#define UA_CMP_LEN	256
 
-typedef struct {
-	struct hlist_head	list;
-	spinlock_t		lock;
-} CliHashBucket;
+static struct {
+	unsigned int	db_size;
+	const char	*db_path;
+	unsigned int	lifetime;
```

Comment:
Maybe I will also rename `life` to` expires` and `tfw_client_set_lifetime` to` tfw_client_set_expires`?